### ISSUE TITLE
stbi_info gives wrong answer for number of components for PNG file (stbi_load is okay)

### ISSUE THREAD

**Issue by mackyle** [nothings/stb#1419]
The test image can be found at [test.png](https://github.com/nothings/stb/raw/6b77d90cc334f1cf/test.png "test image").

1. Download the test.png test image to the current directory
2. Fetch the latest [stb_image.h](https://github.com/nothings/stb/raw/HEAD/stb_image.h) source file
3. Save the source below as "bug.c"
4. Compile with `cc -o bug bug.c`
5. Examine the output:
        $ ./bug  
        stbi_info(test.png): ok=1 x=500 y=300 n=3
        stbi_load(test.png): data=0x100200000 x=500 y=300 n=4
6. Notice that the value for n is 3 from stbi_info but 4 (correct) from stbi_load

``` c
/* bug.c

stb_image.h from (latest) tree of commit 8b5f1f37b5b75829 2022-09-08
test image https://github.com/nothings/stb/raw/6b77d90cc334f1cf/test.png

cc -o bug bug.c  # bug.c is this file

$ ./bug
stbi_info(test.png): ok=1 x=500 y=300 n=3
stbi_load(test.png): data=0x100200000 x=500 y=300 n=4

Notice how the n values are different!

*/

#include <stdio.h>
#define STB_IMAGE_IMPLEMENTATION
#define STBI_NO_THREAD_LOCALS /* not relevant to bug, just more compatible */
#include "stb_image.h"

#define TESTFILE "test.png"

int main()
{
	int x,y,n,ok;
	unsigned char *data;

	ok = stbi_info(TESTFILE, &x, &y, &n);
	printf("stbi_info(%s): ok=%d x=%d y=%d n=%d\n",
		TESTFILE, ok, x, y, n);
	data = stbi_load(TESTFILE, &x, &y, &n, 0);
	printf("stbi_load(%s): data=%p x=%d y=%d n=%d\n",
		TESTFILE, data, x, y, n);

	return 0;
}
```

**Comment by rygorous** [comment]
This bug is now fixed in the dev branch, will be in the next release.

**Comment by rygorous** [comment]
Fixed in 2.28.
